### ISSUE TITLE
Adaptive Hoeffding Tree 'gini' -- 'InactiveLearningNode' object has no attribute 'filter_instance_to_leaves'

### ISSUE THREAD

**Issue by foxriver76** [scikit-multiflow/scikit-multiflow#51]
Used code:
```python
from skmultiflow.evaluation.evaluate_prequential import EvaluatePrequential
from skmultiflow.data.hyper_plane_generator import HyperplaneGenerator
from skmultiflow.trees.hoeffding_adaptive_tree import HAT

"""1. Create stream"""
stream = HyperplaneGenerator(mag_change=0.001, noise_percentage=0.1)

stream.prepare_for_use()

"""2. Create classifier"""
clf = HAT(split_criterion='gini')

"""3. Setup evaluator"""
evaluator = EvaluatePrequential(show_plot=False,
                                pretrain_size=1,
                                max_samples=1000000,
                                metrics=['performance', 'kappa_t', 'kappa_m', 'kappa'],
                                output_file=None)

"""4. Run evaluator"""
evaluator.evaluate(stream=stream, model=clf, model_names=['HAT'])
```

It stops the classification after a few instances with the following error:
```
root - INFO - Prequential Evaluation
root - INFO - Evaluating 1 target(s).
root - INFO - Pre-training on 1 samples.
root - INFO - Evaluating...
root - INFO - Evaluation time: 10.584 s
root - INFO - Total samples: 22444
root - INFO - Global performance:
root - INFO - HAT - Accuracy     : 0.844
root - INFO - HAT - Kappa        : 0.688
root - INFO - HAT - Kappa T      : 0.688
root - INFO - HAT - Kappa M      : 0.692
'InactiveLearningNode' object has no attribute 'filter_instance_to_leaves'
```

best regards

**Comment by jacobmontiel** [comment]
Confirmed. The issue is triggered by the memory management code that was deactivating some leaves and triggered an unhandled condition.

**Comment by jacobmontiel** [comment]
Fixed in e66e4eb94bf1e8d77619ab4c45398b1bfccd1e0f
